sdks/java/core/src/main/java/org/apache/beam/sdk/io/Compression.java

```diff
@@ -240,8 +244,66 @@ public WritableByteChannel writeCompressed(WritableByteChannel channel) throws I
       return Channels.newChannel(
           new SnappyCompressorOutputStream(Channels.newOutputStream(channel), uncompressedSize));
     }
+  },
+
+  /** Four MC Compression Splittable compressed file format leveraging LZ4
+   * Uses LZ4 Fast Compression*/
+  MC4_FAST(".4mc", ".4mc") {
```

Comment:
I believe we will have problems with detection with all of these having the same suffix configurations

Reply:
can you show your experiments verifying that many types with the same exception doesn't break?